docs/store.md

```diff
@@ -0,0 +1,42 @@
+# Pepr Store: A Lightweight Key-Value Store for Pepr Modules
+
+The nature of admission controllers and general watch operations (the `Mutate`, `Validate` and `Watch` actions in Pepr) make some types of complex and long-running operations difficult. There are also times when you need to share data between different actions. While you could manually create your own K8s resources and manager their cleanup, this can be very hard to manage or keep performant at scale. 
```

Comment:
"While you could manually create your own K8s resources and manage~~r~~ their cleanup, this can be very hard to ~~manage or~~ keep performant at scale."

Word stuff.